operator/src/test/java/org/keycloak/operator/KeycloakDeploymentE2EIT.java

```diff
@@ -0,0 +1,138 @@
+package org.keycloak.operator;
+
+import io.fabric8.kubernetes.api.model.EnvVarBuilder;
+import io.quarkus.logging.Log;
+import io.quarkus.test.junit.QuarkusTest;
+import org.awaitility.Awaitility;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Test;
+import org.keycloak.operator.utils.CRAssert;
+import org.keycloak.operator.v2alpha1.crds.Keycloak;
+import org.keycloak.operator.v2alpha1.crds.KeycloakStatusCondition;
+
+import java.time.Duration;
+import java.util.List;
+import java.util.Map;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.keycloak.operator.utils.K8sUtils.deployKeycloak;
+import static org.keycloak.operator.utils.K8sUtils.getDefaultKeycloakDeployment;
+import static org.keycloak.operator.utils.K8sUtils.waitForKeycloakToBeReady;
+
+@QuarkusTest
+public class KeycloakDeploymentE2EIT extends ClusterOperatorTest {
```

Comment:
Considering that the namespace is created and cleaned at the Beginning and End of the Test Class having multiple tests , without knowing the order is not ensuring a deterministic test

Reply:
Why it would not be deterministic? We're cleaning the KC deployment after each test. The namespace should be empty.

Reply:
Yes, we expect the operands to be deleted (and we actually test this). But even if they were not, we would overwrite them with the next KC deployment.